### PR TITLE
Pt 154885468 156004322 custom validation and swagger removal

### PR DESCRIPTION
**Pull request by fabiankrol**
This PR resolves: 
[PT-154885468: Integrate custom validation to cowboy controllers](https://www.pivotaltracker.com/n/projects/2124891/stories/154885468)
[PT-156004322: Remove erlang generated swagger code](https://www.pivotaltracker.com/n/projects/2124891/stories/156004322)

Future improvements to discuss: 
- remove `apps/aehttp/priv/swagger.json` and use `endpoints.erl` to store schema instead.

### CODE REVIEW

**Comment by lucafavatella** [comment]
I see tests failed on CI https://circleci.com/gh/aeternity/epoch/4673
I need to understand which failures - if any - are related to these changes:
```
%%% aecore_forking_SUITE ==> all_nodes.two_nodes.sync_fork_in_wrong_order: FAILED
%%% aecore_forking_SUITE ==> {timeout,
    [{aecore_forking_SUITE,collect_sync_events,1,
         [{file,
              "/home/builder/epoch/_build/test/lib/aecore/test/aecore_forking_SUITE.erl"},
          {line,159}]},
     {aecore_forking_SUITE,sync_fork_in_wrong_order,1,
         [{file,
              "/home/builder/epoch/_build/test/lib/aecore/test/aecore_forking_SUITE.erl"},
          {line,131}]},
     {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1546}]},
     {test_server,run_test_case_eval1,6,
         [{file,"test_server.erl"},{line,1062}]},
     {test_server,run_test_case_eval,9,
         [{file,"test_server.erl"},{line,994}]}]}
```
```
%%% aecore_sync_SUITE ==> all_nodes.two_nodes.start_second_node: FAILED
%%% aecore_sync_SUITE ==> {test_case_failed,
    {retry_exhausted,
        #{info =>
              {470,expect_block,epoch_dev2@localhost,
               {block,1,
                   <<173,144,161,143,82,71,247,64,21,169,150,78,58,247,121,152,
                     208,163,240,137,125,185,215,180,252,4,43,172,40,52,188,92>>,
                   <<167,190,148,131,153,149,64,208,27,9,200,161,25,42,188,121,
                     221,26,148,66,191,236,243,199,28,109,100,177,130,165,171,
                     247>>,
                   <<65,23,82,227,76,23,72,71,84,25,237,55,147,206,72,202,231,
                     19,71,206,244,4,212,227,195,190,88,88,108,164,23,222>>,
                   [{signed_tx,
                        {aetx,aec_coinbase_tx,
                            {coinbase_tx,
                                <<4,179,59,8,248,36,123,146,229,76,137,104,122,
                                  155,224,158,93,74,249,28,58,230,114,103,190,
                                  96,42,111,55,230,147,216,7,83,205,66,169,95,
                                  79,77,218,65,47,69,56,239,186,88,103,140,198,
                                  237,230,159,8,195,35,90,17,43,29,87,114,103,
                                  69>>,
                                1,10}},
                        [<<48,69,2,33,0,141,244,121,62,217,16,104,129,160,
                           75,67,8,33,187,97,38,243,91,195,174,244,240,17,
                           129,198,13,231,209,153,78,136,99,2,32,26,62,221,
                           216,30,121,53,181,62,126,166,79,92,174,172,16,5,
                           117,178,110,109,188,252,137,65,201,35,95,41,251,
                           26,253>>]}],
                   553713663,13739242263053547227,1521545134495,8,
                   [187,524,1983,2041,2568,2865,3457,4383,4954,6090,6236,7749,
                    8157,8242,8557,9227,9651,10408,11033,12352,12651,14548,
                    14934,16308,16467,17025,19055,23147,23654,24781,26323,
                    27427,27873,27937,28764,29077,29627,29733,30452,31366,
                    32564,32740]}},
          prev =>
              {block,0,
                  <<0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
                    0,0,0>>,
                  <<46,116,175,40,171,44,172,167,57,60,232,128,131,35,227,113,
                    214,29,6,134,247,146,53,224,70,53,37,9,176,200,148,82>>,
                  <<0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
                    0,0,0>>,
                  [],553713663,0,0,8,no_value},
          retries => 5,tries => 0}}}
```
```
%%% aecore_sync_SUITE ==> all_nodes.three_nodes.start_second_node: FAILED
%%% aecore_sync_SUITE ==> {test_case_failed,
    {retry_exhausted,
        #{info =>
              {470,expect_block,epoch_dev2@localhost,
               {block,1,
                   <<173,144,161,143,82,71,247,64,21,169,150,78,58,247,121,152,
                     208,163,240,137,125,185,215,180,252,4,43,172,40,52,188,92>>,
                   <<167,190,148,131,153,149,64,208,27,9,200,161,25,42,188,121,
                     221,26,148,66,191,236,243,199,28,109,100,177,130,165,171,
                     247>>,
                   <<80,135,70,243,248,67,124,126,139,172,159,62,28,199,186,
                     125,86,221,10,124,26,236,229,141,188,103,201,156,95,134,
                     39,27>>,
                   [{signed_tx,
                        {aetx,aec_coinbase_tx,
                            {coinbase_tx,
                                <<4,179,59,8,248,36,123,146,229,76,137,104,122,
                                  155,224,158,93,74,249,28,58,230,114,103,190,
                                  96,42,111,55,230,147,216,7,83,205,66,169,95,
                                  79,77,218,65,47,69,56,239,186,88,103,140,198,
                                  237,230,159,8,195,35,90,17,43,29,87,114,103,
                                  69>>,
                                1,10}},
                        [<<48,69,2,33,0,150,198,240,9,134,197,114,92,33,194,
                           40,86,14,214,53,191,47,12,9,91,61,59,62,17,111,
                           214,99,61,34,36,148,124,2,32,47,205,47,129,26,21,
                           71,44,96,180,110,91,38,209,227,181,156,1,52,140,
                           117,178,72,52,123,16,127,24,179,183,77,7>>]}],
                   553713663,16169495089191597681,1521545160868,8,
                   [272,391,734,1987,3678,4596,4635,6692,7791,9125,10119,
                    10545,10810,10959,12359,12400,14225,15065,15978,16053,
                    16415,17429,17571,17701,18799,20872,21249,21328,22372,
                    22717,23638,24630,25746,26202,26752,29134,29449,30059,
                    30305,30320,31091,31872]}},
          prev =>
              {block,0,
                  <<0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
                    0,0,0>>,
                  <<46,116,175,40,171,44,172,167,57,60,232,128,131,35,227,113,
                    214,29,6,134,247,146,53,224,70,53,37,9,176,200,148,82>>,
                  <<0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
                    0,0,0>>,
                  [],553713663,0,0,8,no_value},
          retries => 5,tries => 0}}}
```
```
%%% aecore_sync_SUITE ==> {{case_clause,{badrpc,nodedown}},
 [{aecore_suite_utils,get_node_db_config,1,
                      [{file,"/home/builder/epoch/_build/test/lib/aecore/test/aecore_suite_utils.erl"},
                       {line,112}]},
  {aecore_sync_SUITE,node_db_cfg,1,
                     [{file,"/home/builder/epoch/_build/test/lib/aecore/test/aecore_sync_SUITE.erl"},
                      {line,567}]},
  {aecore_sync_SUITE,'-stop_devs/1-fun-0-',2,
                     [{file,"/home/builder/epoch/_build/test/lib/aecore/test/aecore_sync_SUITE.erl"},
                      {line,137}]},
  {lists,foreach,2,[{file,"lists.erl"},{line,1338}]},
  {aecore_sync_SUITE,stop_devs,1,
                     [{file,"/home/builder/epoch/_build/test/lib/aecore/test/aecore_sync_SUITE.erl"},
                      {line,135}]},
  {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1546}]},
  {test_server,run_test_case_eval1,6,[{file,"test_server.erl"},{line,1147}]},
  {test_server,run_test_case_eval,9,[{file,"test_server.erl"},{line,994}]}]}
```
```
%%% aecore_sync_SUITE ==> {test_case_failed,
    {retry_exhausted,
        #{info =>
              {470,expect_block,epoch_dev2@localhost,
               {block,1,
                   <<173,144,161,143,82,71,247,64,21,169,150,78,58,247,121,152,
                     208,163,240,137,125,185,215,180,252,4,43,172,40,52,188,92>>,
                   <<167,190,148,131,153,149,64,208,27,9,200,161,25,42,188,121,
                     221,26,148,66,191,236,243,199,28,109,100,177,130,165,171,
                     247>>,
                   <<63,104,121,252,36,62,220,204,84,245,231,202,19,218,246,
                     216,241,187,134,135,6,113,118,9,225,9,130,224,107,49,47,
                     110>>,
                   [{signed_tx,
                        {aetx,aec_coinbase_tx,
                            {coinbase_tx,
                                <<4,179,59,8,248,36,123,146,229,76,137,104,122,
                                  155,224,158,93,74,249,28,58,230,114,103,190,
                                  96,42,111,55,230,147,216,7,83,205,66,169,95,
                                  79,77,218,65,47,69,56,239,186,88,103,140,198,
                                  237,230,159,8,195,35,90,17,43,29,87,114,103,
                                  69>>,
                                1,10}},
                        [<<48,70,2,33,0,247,143,143,98,54,155,50,114,76,169,
                           74,182,249,16,242,84,206,158,92,87,178,12,203,
                           122,39,217,56,99,92,179,54,223,2,33,0,206,242,
                           169,102,235,124,91,84,80,95,50,203,91,112,120,
                           217,118,116,58,102,238,128,138,91,66,62,147,71,
                           24,130,112,188>>]}],
                   553713663,7002533059429020765,1521545179947,8,
                   [220,227,2506,2541,2593,2623,3526,3536,3647,4769,5060,5114,
                    5496,5498,5641,5678,8340,8624,9852,10108,13257,14031,
                    14754,15306,17499,18935,18963,19097,20645,21227,21695,
                    21979,23490,25214,25315,26469,26876,27402,29293,30292,
                    31100,32430]}},
          prev =>
              {block,0,
                  <<0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
                    0,0,0>>,
                  <<46,116,175,40,171,44,172,167,57,60,232,128,131,35,227,113,
                    214,29,6,134,247,146,53,224,70,53,37,9,176,200,148,82>>,
                  <<0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
                    0,0,0>>,
                  [],553713663,0,0,8,no_value},
          retries => 5,tries => 0}}}
```

**Comment by lucafavatella** [comment]
@fabiankrol Do you experience locally the following error?
```
%%% aecore_forking_SUITE ==> all_nodes.two_nodes.sync_fork_in_wrong_order: FAILED
```

**Comment by fabiankrol** [comment]
@lucafavatella yes, I'm fixing it

File: apps/aehttp/src/aehttp_api_validate.erl [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+        {ok, NewValue, NewName} -> prepare_param(Rules, NewValue, NewName, Validator);
+        Error -> Error
+    end.
+
+% unused rules
+prepare_param_({"in", _}, _, _, _) -> ok;
+prepare_param_({"name", _}, _, _, _) -> ok;
+prepare_param_({"description", _}, _, _, _) -> ok;
+prepare_param_({"default", _}, _, _, _) -> ok;
+% requred
```
Comment (dincho) [inline apps/aehttp/src/aehttp_api_validate.erl]: very insignificant typo :)

**Comment by fabiankrol** [comment]
@lucafavatella ✅

**Comment by dincho** [comment]
> +345 −8,862
❤️

**Pull request merged**